flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumer.java

```diff
@@ -98,6 +100,11 @@
 	 * shard list retrieval behaviours, etc. */
 	private final Properties configProps;
 
+	/**
+	 * Client configuration for the underlying AWS kinesis client.
+	 */
+	private final ClientConfiguration awsClientConfig;
```

Comment:
Is ClientConfiguration serializable? The entire consumer object will be serialized for deployment.

Reply:
Very good point. It's not :(